yt_idefix/io.py

```diff
@@ -85,3 +87,59 @@ def _read_particle_coords(self, chunks, ptf):
     def _read_particle_fields(self, chunks, ptf, selector):
         # idefix doesn't have particles (yet)
         raise NotImplementedError("Particles are not currently supported for Idefix")
+
+
+class PlutoXdmfIOHandler(BaseIOHandler):
+    _particle_reader = False
+    _dataset_type = "pluto-xdmf"
+
+    def _read_particle_coords(self, chunks, ptf):
+        raise NotImplementedError("Reading of particle fields not yet implemented!")
+
+    def _read_particle_fields(self, chunks, ptf, selector):
+        raise NotImplementedError("Reading of particle fields not yet implemented!")
+
+    def _read_fluid_selection(self, chunks, selector, fields, size):
+        """
+        Filenames are data.<snapnum>.<dbl/flt>.h5
+        <snapnum> needs to be parse from the filename.
+        <snapnum> is the corresponding entry in the <dbl/flt>.h5.out file
+        Example <dbl/flt>.h5.out file:
+            0 0.000000e+00 1.000000e-04 0 single_file little rho vx1 vx2 vx3 prs tr1 tr2 tr3 Temp ndens PbykB mach
+            1 2.498181e+00 3.500985e-03 747 single_file little rho vx1 vx2 vx3 prs tr1 tr2 tr3 Temp ndens PbykB mach
+            2 4.998045e+00 3.400969e-03 1458 single_file little rho vx1 vx2 vx3 prs tr1 tr2 tr3 Temp ndens PbykB mach
+            3 7.497932e+00 3.386245e-03 2186 single_file little rho vx1 vx2 vx3 prs tr1 tr2 tr3 Temp ndens PbykB mach
+        """
+        entry = int(
+            os.path.basename(self.ds.parameter_filename)
+            .replace(".flt.h5", "")
+            .replace("data.", "")
+            .replace(".dbl.h5", "")
+        )
```

Comment:
```suggestion
        if (match := re.search(r"\d{4}", self.ds.basename)) is not None:
            entry = int(match.group())
        else:
            raise RuntimeError(f"Failed to parse entry number from {self.ds.basename}")
```

Reply:
This is much better. Thanks @neutrinoceros !
I had to change `self.ds.basename` to `self.parameter_filename` and it is working.

Reply:
strange. `self.ds.basename` is strictly equivalent to `os.path.basename(self.ds.parameter_filename)`, so I don't get why you needed to switch back. Also, the `parameter_filename` attribute is an historic alias for the `filename` attribute, please use that one instead.

Reply:
(not only is it strange that `basename` doesn't work, it should be alarming; we certainly do not want to accidentally parse a number from a parent directory's name)

Reply:
(Ah, and another thing: there is an implicit rule that threads shouldn't be resolved by the last person who responded, otherwise we just end up hiding stuff from each other instead of making sure we're all on the same page)